### PR TITLE
Currency management in column amount

### PR DESCRIPTION
**Pull request by programmer4web**
1. Rates object, getComparableValue updated ( sorting and filtering are now based on amount value ).  added tests with rates.
2. Configurable 'currency' property used in filters menu, added test.
3. Auto-detect 'currency' property if not configured.
4. If 'autodetect' property is true then 'currency' with be the currency with the highest occurrence ( implemented in function _valuesChanged ). added tests.
5. formatters optimization: when data changes get only formatters that are not already created from the old set of values.
6. Seems there is still an issue, in the input fields, min and max values are not converted to the detected currency.

fixes https://github.com/Neovici/cosmoz-omnitable/issues/24

### CODE REVIEW

**Comment by programmer4web** [comment]
Currency management is ready for review. @nomego

File: cosmoz-omnitable-column-amount.html [hunk 1]
```diff
@@ -49,9 +48,23 @@ <h3 style="margin: 0;">[[ title ]]</h3>
...
+				},
+
+				/**
+				 *  If this is set true then currency property will be the currency with highest occurrence in values
+				 */
+				autodetect: {
+					type: Boolean,
+					value: false
+				},
+				rates: {
```
Comment (plequang) [inline cosmoz-omnitable-column-amount.html]: When a code comment  here to explain what this object should contain

Comment (programmer4web) [reply cosmoz-omnitable-column-amount.html]: Rates object is for currency exchange rates ( In this example Units to EUR is used.)
example is placed in demo rates='{"EUR": 1, "USD":0.8169982616, "AUD":0.6529827192, "SEK": 0.1019271438}'

Data from: 
http://www.xe.com/currencyconverter/convert/?Amount=1&From=USD&To=EUR
http://www.xe.com/currencyconverter/convert/?Amount=1&From=AUD&To=EUR
http://www.xe.com/currencyconverter/convert/?Amount=1&From=SEK&To=EUR

I will add more details in code also.

Comment (plequang) [reply cosmoz-omnitable-column-amount.html]: Yes I was thinking of a code comment that would appear in the api http://127.0.0.1:8081/components/cosmoz-omnitable/#/elements/cosmoz-omnitable-column-amount

**Comment by plequang** [comment]
@programmer4web  Did you see this comment https://github.com/Neovici/cosmoz-omnitable/pull/143#pullrequestreview-101170315 ?

**Comment by programmer4web** [comment]
@plequang yes I will add tests that fails with this case and then an fix for toAmount(value, limit, limitFunc) (I was working on code coverage pull requests)

**Comment by programmer4web** [comment]
@plequang I have added tests for toAmount function with suggested case and updated the function to compare amounts with rates. This is ready for review.

**Comment by programmer4web** [comment]
@plequang Can you review this again please about min, max of input conversion to currency. The actual limiting of input to min, max in in another PR https://github.com/Neovici/cosmoz-omnitable/pull/131

**Comment by programmer4web** [comment]
@plequang I have rebased this. Can you review it again please.

**Pull request merged**